### ISSUE TITLE
Question about 'markers' or 'pin'

### ISSUE THREAD

**Issue by LeoFlandin** [iTowns/itowns#1342]
I'm trying to implement a "marker" or "pin" like feature. That is : 

- A georeferenced image (optional : clamped to the ground)
- always facing the camera
- that have a constant size 

Basically, I'm searching for something like `khtml.maplib.overlay.Marker` in the OpenStreetMap API.
I'm sorry if I don't use the good terms, 3D geospacial is not my area of expertise.

**Comment by zarov** [comment]
What did you try so far ? There are a few examples that may help you:
- [clamping a 3D object to the ground](http://www.itowns-project.org/itowns/examples/#misc_clamp_ground)
- [constant size](http://www.itowns-project.org/itowns/examples/#source_file_gpx_3d): in this example, pins of the GPX track have a constant size once you go past a certain zoom value

Concerning the camera facing, you should take a look at the [FrameRequester](https://github.com/iTowns/itowns/blob/master/src/Core/View.js#L384-L456), to update your pin using the camera (something using `view.camera.camera3d` and the `lookAt` method of your pin (see THREE.js documentation, assuming you are adding a pin like this)).

**Comment by LeoFlandin** [comment]
I did not try anything, I was searching for documentation.
If I understand correctly, there is no built-in function for markers ?
Thanks for the examples, I did try those but I miss the constant size
feature !

**Comment by zarov** [comment]
> If I understand correctly, there is no built-in function for markers ?

I'm not sure what you mean by that, but no. iTowns provides a few helper methods to help in the process, but don't create anything specifically.

> Thanks for the examples, I did try those but I miss the constant size feature !

[This is the lines](https://github.com/iTowns/itowns/blob/master/examples/source_file_gpx_3d.html#L53-L60) for the somewhat constant size, but this is quite complicated for someone outside the project, so here is a brief explanation:

```js
// `this` here is the `THREE.Mesh` object (the pin)
function updatePointScale(renderer, scene, camera) {
    // Get the center of the bounding sphere of the pin.
    point.copy(this.geometry.boundingSphere.center).applyMatrix4(this.matrixWorld);
    
    distance = camera.position.distanceTo(point);
    
    // Get the size of the container (in pixels), which we are using to compute the
    // scale.
    renderer.getSize(size);

    // Clamp the scale between 2 and 100, in your case you can ditch the min/max part.
    // The scale here is a simple ratio between the distance camera-pin, and the y size
    // of the renderer.
    scale = Math.max(2, Math.min(100, distance / size.y));

    // Scale the pin.
    this.scale.set(scale, scale, scale);
    this.updateMatrixWorld();
}
```

**Comment by zarov** [comment]
Sorry, I realised that what you are asking for is being implemented in #1303. If you can wait a bit, you should be able to try it in the next version of itowns !

You can also use [`THREE.Sprite`](https://threejs.org/docs/index.html#api/en/objects/Sprite) in the meantime.

**Comment by LeoFlandin** [comment]
Great ! I'll check it out, thanks.
